e2e/tools/validator/src/validator/cli/__init__.py

```diff
@@ -0,0 +1,170 @@
+# SPDX-FileCopyrightText: 2024-present Sunil Thaha <sthaha@redhat.com>
+#
+# SPDX-License-Identifier: APACHE-2.0
+import click
+import os
+from validator.__about__ import __version__
+from validator.stresser.stresser import ( 
+    run_script,
+)
+from validator.prom_query_validator.prom_query_validator import (
+    PromMetricsValidator, deltas_func, percentage_err
+)
+import yaml
+from typing import NamedTuple
+
+
+import statistics
+
+
+#TODO: decide where to keep the scripts 
+
+
+class Remote(NamedTuple):
+    host: str
+    port: int
+    user: str
+    password: str
+    pkey: str
+
+    def __repr__(self):
+        return f"<Remote {self.user}@{self.host}>"
+
+class VM(NamedTuple):
+    pid: int
+
+class Metal(NamedTuple):
+    vm: VM
+
+class Prometheus(NamedTuple):
+    url: str
+
+class Config(NamedTuple):
+    remote: Remote
+    metal: Metal
+    prometheus: Prometheus
+
+    def __repr__(self):
+        return f"<Config {self.remote}@{self.prometheus}>"
+
+pass_config = click.make_pass_decorator(Config)
+
+
+def load_config(config_file: str) -> Config:
+    """
+    Reads the YAML configuration file and returns a Config object.
+
+    Args:
+        config_file (str): Path to the YAML configuration file.
+
+    Returns:
+        Config: A named tuple containing the configuration values.
+    """
+    with open(config_file, 'r') as file:
+        config = yaml.safe_load(file)
+
+    remote_config = config['remote']
+    remote = Remote(
+        host=remote_config['host'],
+        port=remote_config.get('port', 22),
+        user=remote_config.get('username', 'fedora'),
+        password=remote_config.get('password', None),
+        pkey=os.path.expanduser(remote_config.get('pkey', '~/.ssh/id_rsa')),
+    )
+
+    metal_config = config['metal']
+    vm_config = metal_config['vm']
+    vm = VM( pid=vm_config['pid'],)
+    metal = Metal(vm=vm)
+
+    prometheus_config = config['prometheus']
+    prometheus = Prometheus(
+        url=prometheus_config['url'],
+    )
+
+    return Config(
+        remote=remote, 
+        metal=metal, 
+        prometheus=prometheus,
+    )
+
+
+@click.group(
+    context_settings={"help_option_names": ["-h", "--help"]}, 
+    invoke_without_command=False,
+)
+@click.version_option(version=__version__, prog_name="validator")
+@click.option(
+   "--config-file", "-f", default="validator.yaml",
+    type=click.Path(exists=True),
+)
+@click.pass_context
+def validator(ctx: click.Context, config_file: str):
+    ctx.obj = load_config(config_file)
+
+
+@validator.command()
+@click.option(
+    "--script-path", "-s", 
+    default="scripts/stressor.sh", 
+    type=str,
+)
+@pass_config
+def stress(cfg: Config, script_path: str):
+    PROM_QUERIES = {
+        "vm_process_joules_total": {"name": "kepler_process_package_joules_total", "base_labels": {"job": "metal", "pid": "2093543"}},
```

Comment:
is the pid always 2093543?

Reply:
I think it gets updated here: https://github.com/sustainable-computing-io/kepler/pull/1392/files#diff-23a6e727f292aee124b881c947aa1cfff3bcb9c8a6126d2d131e0a610bfdd8b1R140